### PR TITLE
Cal 7 customers data structure and functions

### PR DESCRIPTION
**Pull request by AlvaroAPansa**
API para customers con las fucnionalidades requeridas.
-Obtener todos los clientes
-Filtrado por ID
-Filtrado por type
-Borrado de customer por ID y sobreescribir JSON

### CODE REVIEW

File: src/data/customers.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[{"id":1,"customer-type":"particular","email":"vgiovannacci0@mozilla.com","buildings":1,"fiscal-address":"0269 Lunder Road"},
```
Comment (francoschiavone) [inline src/data/customers.json]: buildings shouldn't be a list instead of just a number?

**Comment by AlvaroAPansa** [comment]
Closing to correct buidings ID field and separate data and API in two branches

File: src/data/customers.json (hunk 1 above)
Comment (AlvaroAPansa) [reply src/data/customers.json]: Buildings data modified to an array. Thank you!

**Comment by AlvaroAPansa** [comment]
Customers data JSON corrected.

File: src/data/customers.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[{"id":1,"customerType":"particular","email":"ecallingham0@ebay.com","buidingsId":[100,8,45,36,4,99,23,55,18,10],"fiscalAddress":"5260 Mallory Crossing"},
```
Comment (wgnr) [inline src/data/customers.json:1]: I've seen that there're some field missing.
Ex:
- Name
- Contact Name
- Phone

Is this intentional? May be you are just showing that the business logic is working.

File: src/data/customers.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+{"id":2,"customerType":"particular","email":"hvanyukhin1@purevolume.com","buidingsId":[70,93,39,51,86,28,47,76,64],"fiscalAddress":"8 Donald Lane"},
```
Comment (wgnr) [inline src/data/customers.json:2]: `buidingsId:[70,93,39,51,86,28,47,76,64]`
For now this gonna be OK. Later on, when we will be designing our DB relationship schema, we'll remove that field to simply add in table Buildings the  "id_customer".

File: src/controllers/customers.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  }
+});
+
+// Delete customer (by id)
+router.delete('/', (req, res) => {
+  //console.log('id: '+req.query.id);
+  const idFound = customers.some(customer => customer.id === parseInt(req.query.id));
+  if (idFound) {
+    let customersDeleted = customers.filter(customer => customer.id !== parseInt(req.query.id));
+    res.json({
```
Comment (wgnr) [inline src/controllers/customers.js:37]: I would response the client that deletion was successful once the JSON was written in fs.
(inside `else` line 46)

**Review by wgnr — APPROVED** [review]

**Pull request merged**